src/rhsm/config.py

```diff
@@ -120,8 +121,10 @@ def __init__(self, config_file=None, defaults=None):
 
     def save(self, config_file=None):
         """Writes config file to storage."""
-        fo = open(self.config_file, "w")
+        fo = tempfile.NamedTemporaryFile(mode="w", dir="/etc/rhsm", delete=False)
         self.write(fo)
+        fo.close()
```

Comment:
I prefer to use `with` statement. When some exception happens inside `with` statement, then the file is closed automatically.

Reply:
Done.